tests/imputers/test_ModeImputer.py

```diff
@@ -159,201 +134,22 @@ def test_fit_not_changing_data_weighted(self):
-
-        return df
-
-    def expected_df_3():
-        """Expected output for test_nulls_imputed_correctly_3."""
-        df = d.create_df_9()
-
-        for col in ["a"]:
-            df.loc[df[col].isna(), col] = 6
-
-        return df
-
-    def test_check_is_fitted_called(self, mocker):
-        """Test that BaseTransformer check_is_fitted called."""
-        df = d.create_df_1()
-
-        x = ModeImputer(columns="a")
-
-        x.fit(df)
-
-        expected_call_args = {0: {"args": (["impute_values_"],), "kwargs": {}}}
-
-        with ta.functions.assert_function_call(
-            mocker,
-            tubular.base.BaseTransformer,
-            "check_is_fitted",
-            expected_call_args,
-        ):
-            x.transform(df)
-
-    def test_super_transform_called(self, mocker):
-        """Test that BaseTransformer.transform called."""
-        df = d.create_df_1()
-
-        x = ModeImputer(columns="a")
-
-        x.fit(df)
-
-        expected_call_args = {0: {"args": (d.create_df_1(),), "kwargs": {}}}
-
-        with ta.functions.assert_function_call(
-            mocker,
-            tubular.base.BaseTransformer,
-            "transform",
-            expected_call_args,
-        ):
-            x.transform(df)
-
-    @pytest.mark.parametrize(
-        ("df", "expected"),
-        ta.pandas.adjusted_dataframe_params(d.create_df_3(), expected_df_1()),
-    )
-    def test_nulls_imputed_correctly(self, df, expected):
-        """Test missing values are filled with the correct values."""
-        x = ModeImputer(columns=["a", "b", "c"])
-
-        # set the impute values dict directly rather than fitting x on df so test works with helpers
-        x.impute_values_ = {"a": 1.0, "b": 1.0, "c": -6.0}
-
-        df_transformed = x.transform(df)
-
-        ta.equality.assert_equal_dispatch(
-            expected=expected,
-            actual=df_transformed,
-            msg="Check nulls filled correctly in transform",
-        )
-
-    @pytest.mark.parametrize(
-        ("df", "expected"),
-        ta.pandas.adjusted_dataframe_params(d.create_df_3(), expected_df_2()),
-    )
-    def test_nulls_imputed_correctly_2(self, df, expected):
-        """Test missing values are filled with the correct values - and unrelated columns are not changed."""
-        x = ModeImputer(columns=["a"])
-
-        # set the impute values dict directly rather than fitting x on df so test works with helpers
-        x.impute_values_ = {"a": 1.0}
-
-        df_transformed = x.transform(df)
-
-        ta.equality.assert_equal_dispatch(
-            expected=expected,
-            actual=df_transformed,
-            msg="Check nulls filled correctly in transform",
-        )
-
-    @pytest.mark.parametrize(
-        ("df", "expected"),
-        ta.pandas.row_by_row_params(d.create_df_9(), expected_df_3())
-        + ta.pandas.index_preserved_params(d.create_df_9(), expected_df_3()),
-    )
-    def test_nulls_imputed_correctly_3(self, df, expected):
```

Comment:
worth reintroducing this one for the time being too

Reply:
No probs - had to add the expected df creation in to the base tests file

Reply:
thanks! is there a reason to add the expected df to base tests instead of the mode tests file?

Reply:
Just where the other ones seem to live! I guess if this is only going to be used by mode I can move it back?

Reply:
yeahh bit of a nitpick but think that makes more sense to me, if OK!